subsys/emds/emds.c

```diff
@@ -253,7 +289,17 @@ int emds_prepare(void)
 	(void)emds_entries_size(&size);
 
 	rc = emds_flash_prepare(&emds_flash, size);
-	if (rc) {
+
+	/** If flash is empty after prepare, imediatley make an
+	 * extra copy of the current entries as backup before the
+	 * next EMDS event in case it should fail
+	 */
+	if (rc == 1) {
```

Comment:
This is not necessary if not EMDS_AUTO_RECOVER is configured. Reduce number of store calls.